Source/BSN.Commons/Infrastructure/MessageBroker/EventAggregator/InMemoryEventAggregatorSubscriptionManager.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator.Contracts;
+using IEvent = BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator.Contracts.IEvent;
+
+namespace BSN.Commons.Infrastructure.MessageBroker.EventAggregator
+{
+    public partial class InMemoryEventAggregatorSubscriptionManager : IEventAggregatorSubscriptionManager
+    {
+        public InMemoryEventAggregatorSubscriptionManager()
+        {
+            _eventReceivers = new Dictionary<string, IEventReceiver>();
+            _eventsType = new List<Type>();
+        }
+
+        public void AddSubscription<TEvent>(IEventReceiver eventReceiver) where TEvent : IEvent
+        {
+            string eventName = GetEventKey<TEvent>();
+
+            DoAddSubscription(eventReceiver, eventName);
+
+            if (!_eventsType.Contains(typeof(TEvent)))
+            {
+                _eventsType.Add(typeof(TEvent));
+            }
+        }
+
+        private void DoAddSubscription(IEventReceiver eventReceiver, string eventName)
```

Comment:
Name of this method is strange.

Please make better naming

Reply:
done.